src/hooks/useQuizTrackingAndCbEvents.ts

```diff
@@ -0,0 +1,183 @@
+        quizApiState?.quizResults.response
+      ) {
+        const {
+          quiz_id,
+          quiz_session_id,
+          quiz_version_id,
+          request: { section },
+        } = quizApiState.quizResults;
+
+        cioClient?.tracker.trackQuizConversion({
+          quiz_id,
+          quiz_version_id,
+          quiz_session_id,
+          item_id: result.data?.id,
+          item_name: result.value,
+          section,
+          variation_id: result.data?.variation_id,
+          revenue: (price && String(price)) || undefined,
+        });
+      }
+
+      // User custom callback function
+      if (isFunction(onAddToCartClick)) {
+        e.stopPropagation();
+        onAddToCartClick(result);
+      }
+    },
+    [quizApiState, cioClient, onAddToCartClick]
+  );
+
+  // Quiz result click tracking event
+  const resultClickHandler = useCallback(
+    (result: Partial<GetBrowseResultsResponseData>, position: number) => {
+      if (
+        quizApiState?.quizResults &&
+        quizApiState?.quizResults.request &&
+        quizApiState?.quizResults.response
+      ) {
+        const {
+          quiz_id,
+          quiz_session_id,
+          quiz_version_id,
+          result_id,
+          request: { section, page, num_results_per_page },
+          response: { total_num_results },
+        } = quizApiState.quizResults;
+
+        cioClient?.tracker.trackQuizResultClick({
+          quiz_id,
+          quiz_version_id,
+          quiz_session_id,
+          item_id: result.data?.id,
+          item_name: result?.value,
+          section,
+          result_count: total_num_results,
+          result_page: page,
+          result_id,
+          result_position_on_page: position,
+          num_results_per_page,
+        });
+      }
+
+      // User custom callback function
+      if (onQuizResultClick && isFunction(onQuizResultClick)) {
+        onQuizResultClick(result, position);
+      }
+    },
+    [quizApiState, cioClient, onQuizResultClick]
+  );
+
+  // Quiz results loaded
+  useEffect(() => {
+    if (quizApiState.quizResults) {
+      // User custom callback function
+      if (onQuizResultsLoaded && isFunction(onQuizResultsLoaded)) {
+        onQuizResultsLoaded(quizApiState.quizResults);
+      }
+
+      // Tracking call
+      const { quiz_id, quiz_session_id, quiz_version_id, result_id, request, response } =
+        quizApiState.quizResults;
+
+      cioClient?.tracker.trackQuizResultsLoaded({
+        quiz_id,
+        quiz_version_id,
+        quiz_session_id,
+        url: window.location.href,
+        section: request?.section,
+        result_count: response?.total_num_results,
+        result_page: request?.page,
+        result_id,
+      });
+    }
+  }, [quizApiState, cioClient, onQuizResultClick, onQuizResultsLoaded]);
+
+  return {
+    addToCartClickHandler,
+    resultClickHandler,
+    quizNextHandler,
+    quizBackHandler,
+  };
+};
+
+export default useQuizTrackingAndCbEvents;
```

Comment:
This file name is already long so would it make sense to make it `useQuizTrackingAndCallbackEvents`?